### PR TITLE
feat: access client side logging from Test Tools Modal

### PR DESCRIPTION
**Pull request by TMisiukiewicz**
### Fixed Issues

$ https://github.com/Expensify/App/issues/38456
PROPOSAL:

### iOS/Android:
Open dev menu and select "Open Test Preferences"

### iOS Web/Android Web
Use physical device and use four-finger tap or switch `isVisible={!!isTestToolsModalOpen}` to `isVisible` in `src/components/TestToolsModal.tsx`

1. Enable "Client side logging"
2. Wait a while and disable "Client side logging"

**iOS/Android**
3. Verify path of the file is visible
4. Press "Share" and verify sharing modal appeared
5. Go to the Files on the device and verify it was saved

**iOS Web/Android Web**
3. Verify file was downloaded on the device

### All platforms - access from Troubleshoot page 
1. Go to Profile -> About -> Troubleshoot
2. Enable Client Side Logging
3. Disable Client Side Logging

**iOS/Android**
4. Verify the filepath is displayed and Share button is visible
5. Press "Share" and verify the Share menu is visible

**Other platforms**
4. Verify file was downloaded to the device

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open the app
2. Tap any area of the app with four fingers simultaneously.
3. Verify "Client side logging" is available
4. Enable Client Side logging
7. Disable Client Side logging

**iOS/Android**
8. Verify path of the file is visible
9. Press "Share" and verify sharing modal appeared
10. Go to the Files on the device and verify it was saved

**iOS Web/Android Web**
6. Verify file was downloaded on the device

### All platforms - access from Troubleshoot page 
1. Go to Profile -> About -> Troubleshoot
2. Enable Client Side Logging
3. Disable Client Side Logging

**iOS/Android**
4. Verify the filepath is displayed and Share button is visible
5. Press "Share" and verify the Share menu is visible

**Other platforms**
4. Verify file was downloaded to the device
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/13985840/5f103f8b-74a4-41ea-9d38-09bd673232d6

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/13985840/4856963b-7f8e-41fb-aa73-016fafe11d04

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/13985840/4ee76ce6-ef92-42ed-a508-42aa30ac56a2

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/13985840/37c81180-ac77-4f1a-a483-9ecba34ef483

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/13985840/a5c906f6-5159-48b8-b9ac-7cddf117f6a2

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/13985840/296014fc-6d32-49d5-bec9-082f19f918a8

</details>

### CODE REVIEW

File: src/components/ClientSideLoggingToolMenu/index.native.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import type {ClientSideLoggingToolMenuOnyxProps, ClientSideLoggingToolProps} from './types';
+
+function ClientSideLoggingToolMenu({shouldStoreLogs, capturedLogs}: ClientSideLoggingToolProps) {
+    const [file, setFile] = useState<{path: string; newFileName: string; size: number}>();
+
+    const onToggle = () => {
+        if (!shouldStoreLogs) {
+            Console.setShouldStoreLogs(true);
+        } else {
+            if (capturedLogs) {
```
Comment (adhorodyski) [inline src/components/ClientSideLoggingToolMenu/index.native.tsx:23]: these could be simplified with early returns on all the edge cases (like no logs captured)

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                />
+            </TestToolRow>
+            {!!file && (
+                <>
+                    <Text style={[styles.textLabelSupporting, styles.mb4]}>{`path: ${file.path}`}</Text>
+                    <TestToolRow title="Logs">
+                        <Button
+                            small
+                            text="Share"
+                            onPress={onShareLogs}
```
[the remark this answers is not in the document]
Comment (TMisiukiewicz) [reply src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx:37]: this has to be platform-specific, as there is no share option for web/desktop

File: src/components/ClientSideLoggingToolMenu/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ClientSideLoggingToolMenu.displayName = 'ClientSideLoggingToolMenu';
+
+export default withOnyx<ClientSideLoggingToolProps, ClientSideLoggingToolMenuOnyxProps>({
+    capturedLogs: {
+        key: ONYXKEYS.LOGS,
+    },
+    shouldStoreLogs: {
+        key: ONYXKEYS.SHOULD_STORE_LOGS,
+    },
+})(ClientSideLoggingToolMenu);
```
[the remark this answers is not in the document]
Comment (TMisiukiewicz) [reply src/components/ClientSideLoggingToolMenu/index.tsx:34]: thanks, now i am subscribing to Onyx only in `ClientSideLogginToolMenu` and passing it down to base component

**Comment by TMisiukiewicz** [comment]
@s77rt thanks for your review, I have addressed all of your comments

File: src/components/ClientSideLoggingToolMenu/types.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import type {OnyxEntry} from 'react-native-onyx';
+import type {Log} from '@libs/Console';
+
+type CapturedLogs = Record<number, Log>;
```
[the remark this answers is not in the document]
Comment (s77rt) [reply src/components/ClientSideLoggingToolMenu/types.ts]: Change
```js
[ONYXKEYS.LOGS]: Record<number, OnyxTypes.Log>;
```
to:
```js
[ONYXKEYS.LOGS]: OnyxTypes.CapturedLogs;
```

File: src/components/ClientSideLoggingToolMenu/index.android.tsx [hunk 5]
```diff
@@ -1,47 +1,31 @@
...
+        localFileCreate('logs', JSON.stringify(logs, null, 2)).then((localFile) => {
+            RNFetchBlob.MediaCollection.copyToMediaStore(
+                {
+                    name: localFile.newFileName,
+                    parentFolder: '',
+                    mimeType: 'text/plain',
+                },
+                'Download',
+                localFile.path,
+            );
```
Comment (s77rt) [inline src/components/ClientSideLoggingToolMenu/index.android.tsx:25]: Is there a reason to copy the files to the media store?

File: src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (s77rt) [reply src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx:57]: Let's pass the file prop through onShareLogs for now

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
On toggle on I think we need to hide the previous path and share buttons

https://github.com/Expensify/App/assets/16493223/3b56df3e-0113-4e6a-92b1-6edc043a5a65

File: src/components/ClientSideLoggingToolMenu/index.android.tsx (hunk 5 above)
Comment (TMisiukiewicz) [reply src/components/ClientSideLoggingToolMenu/index.android.tsx:25]: yes, on Android it is required to copy the file to media store to make it available in the public download directory

**Comment by s77rt** [comment]
@shawnborton The path (in muted text) and the Logs menu item

**Comment by shawnborton** [comment]
Got it - yeah, I think it works.

**Comment by quinthar** [comment]
Who is this waiting on -- @s77rt ?  what is the next step and when will it happen?

**Comment by TMisiukiewicz** [comment]
> Not sure I can test on Android. It does not seem to be working (file not being saved?)

I think it might be an emulator thing, here's the video from Android device:

https://github.com/Expensify/App/assets/13985840/eb715be6-f3d0-4db6-88b4-c9ee4cdd3925

On the emulator I have the same issue, but i think we might assume the file is accessible by having a file name here:

https://github.com/Expensify/App/assets/13985840/1dd81cec-e9d2-46eb-94a9-bbdb7d11a1aa

**Comment by TMisiukiewicz** [comment]
> Pressing Share causes the illustration to disappear

Seems like it's related to the #38093. When Share menu is displayed, the app state is changed to backgrounded. I couldn't find the way to determine if share menu is open or not. Do you think it is a blocker or we could proceed without it and try to find a solution later?

File: src/components/ClientSideLoggingToolMenu/types.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    shouldStoreLogs: OnyxEntry<boolean>;
+};
+
+type BaseClientSideLoggingToolProps = {
+    /** Locally created file */
+    file?: {path: string; newFileName: string; size: number};
+    /** Action to run when pressing Share button */
+    onShareLogs?: () => void;
+    /** Action to run when toggling the switch */
+    onToggleSwitch: (logs: Log[]) => void;
```
Comment (s77rt) [inline src/components/ClientSideLoggingToolMenu/types.ts:18]: Rename to onDisableLogging

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
> I think it might be an emulator thing, here's the video from Android device

Can you confirm if the file is actually there? e.g. was able to read the content or share it

**Comment by s77rt** [comment]
> Do you think it is a blocker or we could proceed without it and try to find a solution later?

Thanks for checking this one. It's not a blocker, we can proceed and fix it later

**Comment by TMisiukiewicz** [comment]
> @TMisiukiewicz Can you please address this too [#38803 (comment)](https://github.com/Expensify/App/pull/38803#discussion_r1537671565)

@s77rt could you elaborate more on that? because I'm not sure I understand what exactly needs to be done, do you want to use both `file` prop and pass `file` through `onShareLog` param?

**Comment by s77rt** [comment]
@TMisiukiewicz Can you please unify the testing steps? I think this is testable on all platforms

File: src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (s77rt) [reply src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx:75]: Actually there is no benefit in doing this.

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16493223/c8a57133-8280-433a-9372-bb1f04280c03


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16493223/3ede9486-47b6-48e4-873b-c88a2afb7b5d


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16493223/97b30227-fb64-4efd-bf4e-5270170ed976


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16493223/6775a980-79d2-4947-ad3a-38603954ec7f


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/16493223/1ac3453d-8a7a-46d7-b0f2-07a78d1da52c


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16493223/c358bbc0-4f00-4701-bd05-44b6de827184


</details>

**Comment by TMisiukiewicz** [comment]
> > I think it might be an emulator thing, here's the video from Android device
> 
> Can you confirm if the file is actually there? e.g. was able to read the content or share it

I just tested it by sending it with an email, seems like a file is visible as an attachment and once sent, it's possible to open it and read the content:

https://github.com/Expensify/App/assets/13985840/7f13b44a-b36c-4c27-a7e6-cb3297e78b71

https://github.com/Expensify/App/assets/13985840/ab1bde3e-0d00-444c-b32a-5d87ac226cea

I've updated test steps for Web & desktop for all platforms together with missing recordings for web & desktop based on Troubleshoot section

**Comment by s77rt** [comment]
@TMisiukiewicz Thank you!

**Review by s77rt — APPROVED** [review]

**Review by s77rt — APPROVED** [review]

File: src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        const logs = Object.values(capturedLogs);
+        const logsWithParsedMessages = parseStringifyMessages(logs);
+
+        onDisableLogging(logsWithParsedMessages);
+        Console.disableLoggingAndFlushLogs();
+    };
+    const styles = useThemeStyles();
+    return (
+        <>
+            <TestToolRow title="Client side logging">
```
Comment (amyevans) [inline src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx:61]: Should we add a translation for this phrase? I see that it's consistent with other titles in Tools (all are hardcoded English) so NAB, but curious if there's a reason for not translating

File: src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                    onToggle={onToggle}
+                />
+            </TestToolRow>
+            {!!file && (
+                <>
+                    <Text style={[styles.textLabelSupporting, styles.mb4]}>{`path: ${file.path}`}</Text>
+                    <TestToolRow title="Logs">
+                        <Button
+                            small
+                            text="Share"
```
Comment (amyevans) [inline src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx:74]: There's an existing translation for this, let's use it

File: src/components/ProfilingToolMenu/index.native.tsx [hunk 9]
```diff
@@ -142,29 +142,24 @@ function ProfilingToolMenu({isProfilingInProgress = false}: ProfilingToolMenuPro
...
-                        text="Share"
-                        onPress={onDownloadProfiling}
-                    />
-                </TestToolRow>
+                <>
+                    <Text style={[styles.textLabelSupporting, styles.mb4]}>{`path: ${pathIOS}`}</Text>
+                    <TestToolRow title="Profile trace">
+                        <Button
+                            small
+                            text="Share"
```
Comment (amyevans) [inline src/components/ProfilingToolMenu/index.native.tsx:158]: Same comment as above

File: src/libs/Console/index.ts [hunk 10]
```diff
@@ -118,5 +119,29 @@ function createLog(text: string) {
     }
 }
 
-export {sanitizeConsoleInput, createLog, shouldAttachLog};
+/**
+ * Loops through all the logs and parses the message if it's a stringified JSON
+ * @param logs Logs captured on the current device
+ * @returns CapturedLogs with parsed messages
+ */
+function parseStringifyMessages(logs: Log[]): Log[] {
```
Comment (amyevans) [inline src/libs/Console/index.ts:127]: Perhaps `Stringified` instead?
```suggestion
function parseStringifiedMessages(logs: Log[]): Log[] {
```

**Review by amyevans — COMMENTED** [review]
Love this feature improvement, and tests well! Just a few minor thoughts on the code

**Comment by TMisiukiewicz** [comment]
@amyevans thanks for you comments! I have updated the translations, I think it makes sense to use translation for Client side logging as well, would be good to remove all existing hard-coded phrases in the future

**Comment by TMisiukiewicz** [comment]
@amyevans thank you, updated!

**Review by amyevans — APPROVED** [review]

**Comment by jayeshmangwani** [comment]
Hi 👋  @TMisiukiewicz @s77rt  ,For Client side logging, Why are we showing the share option for Android and iOS only? I am reviewing a PR that captures profiling data for web/desktop, and there we are showing the share button, but it downloads the file on the web. Moving forward with that PR, I need to understand why we hide the Share button for web/desktop for Client side logging.

**Comment by s77rt** [comment]
@jayeshmangwani On web there is no share functionality. The log file is downloaded automatically after you stop capturing the logs.

**Pull request merged**